lib/src/enforcer.dart

```diff
@@ -8,5 +11,38 @@ class Enforcer extends CoreEnforcer {
   /// [policyFile] is the path of the policy file.
   /// [adapter] is the adapter.
   /// [enableLog] whether to enable Casbin's log.
-  Enforcer(var model, String policyFile, Adapter adapter, bool enableLog);
+  static final Enforcer _enforcer = Enforcer._();
+
+  factory Enforcer._() => _enforcer;
```

Comment:
I'm not sure what you're trying to do here, shouldn't you call `Enforcer()` on line 14? Seems like `Enforcer._()` does nothing so it can be deleted.

Reply:
It just returns the singleton as of now. I thought we could add more initialisation logic inside the private constructor in the future. Anyways, I'll remove it for now. 
Yrs, I'll add the test cases. @KNawm

Reply:
I see, yeah if we need it in the future it can always be added again.